src/event/inventory/PlayerEnchantmentOptionsRequestEvent.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\event\inventory;
+
+use pocketmine\block\inventory\EnchantInventory;
+use pocketmine\event\Cancellable;
+use pocketmine\event\CancellableTrait;
+use pocketmine\event\Event;
+use pocketmine\item\enchantment\EnchantmentOption;
+use pocketmine\player\Player;
+use pocketmine\utils\Utils;
+
+/**
+ * Called when a player inserts an item into an enchanting table's input slot.
+ * The options provided by the event will be shown on the enchanting table menu.
+ */
+class PlayerEnchantmentOptionsRequestEvent extends Event implements Cancellable{
```

Comment:
It can extends `InventoryEvent` ?

Reply:
@dktapps said that inventory is not a related subject, so no, it's better not to extend InventoryEvent